pkg/core/mpt/helpers.go

```diff
@@ -3,30 +3,22 @@ package mpt
 // lcp returns longest common prefix of a and b.
 // Note: it does no allocations.
 func lcp(a, b []byte) []byte {
-	if len(a) < len(b) {
-		return lcp(b, a)
+	m := len(a)
+	if len(b) < m {
+		m = len(b)
 	}
 
 	var i int
-	for i = 0; i < len(b); i++ {
-		if a[i] != b[i] {
-			break
-		}
+	for ; i < m && a[i] == b[i]; i++ {
 	}
 
 	return a[:i]
 }
 
 func lcpMany(kv []keyValue) []byte {
-	if len(kv) == 1 {
-		return kv[0].key
-	}
-	p := lcp(kv[0].key, kv[1].key)
-	if len(p) == 0 {
-		return p
-	}
-	for i := range kv[2:] {
-		p = lcp(p, kv[2+i].key)
+	p := kv[0].key
+	for i := 1; i < len(kv); i++ {
+		p = lcp(p, kv[i].key)
```

Comment:
Add a shortcut for `len(p) == 0`? Was this case tested? Where is LCP benchmark?

Reply:
I have removed this commit after benchmarking again, on a clean machine it showed no meaningful improvements.